File: Programming-languages-and-laboratory/Java/Lessons/34_20221221/29-soluzione-a/Pyhton/dipendente.py
class Dipendente:
    codice: int
    tipo: str
    nome: str
    costo_orario: float # Python non ha il tipo double

    servizi: list

    def __init__(self, codice: int, tipo: str, nome: str, costo_orario: float, servizi: list = None):
        self.codice = codice
        self.tipo = tipo
        self.nome = nome
        self.costo_orario = costo_orario

        if servizi is None:
            self.servizi = []
        else:
            self.servizi = servizi

    def add_servizio(self, servizio):
        self.servizi.append(servizio)

    def __str__(self):
        tipo = self.tipo
        if self.tipo == "nutrizionista":
            tipo = "nutri"
        return self.nome + ", " + str(self.codice) + ", " + tipo

File: II - Linguaggi di programmazione e laboratorio/Java/Exercises/41_compito_java_2022_11_10/Dipendente.java
public class Dipendente {
	protected int codice;
	protected String tipo;
	protected String nomeDipendente;
	protected float costoOrario;
	protected int numeroServiziSvolti;

	public Dipendente(int codice, String tipo, String nomeDipendente, float costoOrario) {
		this.codice = codice;
		this.tipo = tipo;
		this.nomeDipendente = nomeDipendente;
		this.costoOrario = costoOrario;
		this.numeroServiziSvolti = 0;		
	}
	
	public void incrementaNumeroServiziSvolti() {
		numeroServiziSvolti++;
	}

	public String toString() {
		return nomeDipendente + "\t" + codice;
	}
}